### PR TITLE
CVE-2016-1682 and CVE-2013-0925

### PR DESCRIPTION
**Pull request by rh3547**
Initial update of CVEs CVE-2016-1682 and CVE-2013-0925. CVE-2013-0925 is somewhat incomplete but is as filled out as I can get it at the moment.

### CODE REVIEW

File: cves/CVE-2013-0925.yml [hunk 1]
```diff
@@ -55,15 +58,19 @@ fixes:
   :note: ''
 vccs:
 - :commit:
-  :note: 
+  :note: |
+    I can't for the life of me find the vcc. There are a few commits that claim
+    to have "fixed" the same bug, but they all look very similar and simply
+    tweak the code a little bit. I wasn't able to find exactly what commit
+    introduced the vulnerability, even after looking back several layers deep.
```
Comment (mesternefeld) [inline cves/CVE-2013-0925.yml]: Maybe include one of those? I know you can look on github if that makes it easier. Put the commit hash with the farthest back you can go. :)

File: cves/CVE-2013-0925.yml [hunk 2]
```diff
@@ -94,20 +101,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: Unable to tell because I can't seem to find the origin of the vulnerability.
+  date:
```
Comment (mesternefeld) [inline cves/CVE-2013-0925.yml]: there should be a bug report from Chromium somewhere for this vulnerability. There was a bug report files for this vulnerability. If you look through the links here, the bug report was indeed linked. https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2013-0925

File: cves/CVE-2013-0925.yml [hunk 3]
```diff
@@ -153,38 +169,38 @@ lessons:
...
   yagni:
-    applies: 
-    note: 
+    applies: false
+    note:
   complex_inputs:
-    applies: 
-    note: 
+    applies: false
+    note:
```
Comment (mesternefeld) [inline cves/CVE-2013-0925.yml]: It's a little hard to believe there were no lessons, especially if attackers can gain access to tabs. Think about it again.

File: cves/CVE-2016-1682.yml [hunk 4]
```diff
@@ -154,38 +160,38 @@ lessons:
...
-    applies: 
-    note: 
+    applies: false
+    note:
   complex_inputs:
-    applies: 
-    note: 
+    applies: false
+    note:
 mistakes:
```
Comment (mesternefeld) [inline cves/CVE-2016-1682.yml]: Again, seems strange how there were no lessons to be learned here.

File: cves/CVE-2013-0925.yml [hunk 5]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: CWE-358
```
Comment (wylaris) [inline cves/CVE-2013-0925.yml]: Some sites say this CVE is a CWE 358 but this one might be more applicable.

File: cves/CVE-2016-1682.yml [hunk 6]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Checks to verify the content security policy (CSP) of a request were missing
+  from a function that registers service workers. This allowed attackers to
+  bypass CSP and register service workers even if they shouldn't have been
```
Comment (bencrossgrove) [inline cves/CVE-2016-1682.yml]: Can you elaborate on what a 'service worker' is?

File: cves/CVE-2016-1682.yml [hunk 7]
```diff
@@ -198,4 +204,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    It seems as though the developer just didn't think to check for CSP
+    before registering the service workers. The fix was simply adding an if
```
Comment (bencrossgrove) [inline cves/CVE-2016-1682.yml]: Very interesting that the fix was so simple. Just goes to show that anyone can make these seemingly 'simple' mistakes!

**Pull request merged**